supercollider/server.py

```diff
@@ -13,188 +16,134 @@ def __init__(self, hostname="127.0.0.1", port=57110):
+        return self._await_response("/g_queryTree.reply", callback=self.simple_handler)
+    
+    def get_status(self):
         self._send_msg("/status")
-        if blocking:
-            return self._await_response("/status.reply", None, _handler)
-        elif callback:
-            self._add_handler("/status.reply", None, _handler)
 
-    def get_version(self, callback=None, blocking=True):
-        """
-        Returns the current Server version.
-
-        Example:
-            >>> server.version
-            {
-                'program_name': "scsynth",
-                'version_major': 3,
-                'version_minor': 10,
-                'version_patch': ".3",
-                'git_branch': "HEAD",
-                'commit_hash': "67a1eb18"
-            }
-        """
-        def _handler(args):
-            args_dict = {
-                "program_name": args[0],
-                "version_major": args[1],
-                "version_minor": args[2],
-                "version_patch": args[3],
-                "git_branch": args[4],
-                "commit_hash": args[5]
-            }
-            if callback:
-                callback(args_dict)
-            return args_dict
+        return self._await_response("/status.reply", None, self.status_handler)
 
+    def get_version(self):
         self._send_msg("/version")
-        if blocking:
-            return self._await_response("/version.reply", None, _handler)
-        elif callback:
-            self._add_handler("/version.reply", None, _handler)
 
-    def _send_msg(self, msg, *args):
-        liblo.send(self.client_address, msg, *args)
+        return self._await_response("/version.reply", None, version_handler)
 
-    def _await_response(self, address, match_args=(), callback=lambda rv=None: rv):
-        event = threading.Event()
-        rv = None
+    # ------------------- Callback Timeout Logic ------------------- #
+    def _await_response(self, address, match_args=(), callback=None):
 
-        def callback_with_timeout(*args):
-            event.set()
+        rv = None
+        # Sets the thread event when message is received
+        # It also overwrites the callback (dispatcher) function to capture the return value.
+        # This is necessary because the dispatcher handler can't return anything.
+        # Make sure to unmap the callback before it is overwritten to avoid duplicate callbacks.
+        # TODO: Find a better way to do this
+        def _callback_with_timeout(*args):
+            self.event.set()
             if callback:
                 nonlocal rv
-                rv = callback(*args)
+                rv = callback(address, *args[1:])
+
+        self.dispatcher.map(address, _callback_with_timeout)
 
-        self._add_handler(address, match_args, callback_with_timeout)
-        responded_before_timeout = event.wait(globals.RESPONSE_TIMEOUT)
+        responded_before_timeout = self.event.wait(0.25)
```

Comment:
Bad practice to have magic numbers in the code (the original used `globals.RESPONSE_TIMEOUT` - could this still be used?)

Reply:
Absolutely. I removed it during the refactoring process and forgot to put it back